### ISSUE TITLE
Pressing `enter` to submit form behaves strangely when using custom inputs

### ISSUE THREAD

**Issue by cmackenz** [adfinis-sygroup/ember-validated-form#104]
I have a form that has a custom input and when using the `enter` key to submit the form I get some weird results.  

First time pressing enter when focused on the `Reason` input my custom input's value seems to get cleared out and the form does not submit. 

Pressing enter again submits the form with the custom input's value which was selected before pressing the enter key the first time.

If I submit the form by clicking on the submit button the form behaves as expected.

If I remove the custom input and press enter to submit the form it behaves as expected.  

``` hbs
{{#f.input name='serviceable' class="serviceable" as |fi|}}
  {{#unless hideServiceable}}
    <div class="required field">
      <label>Resource</label>
      {{serviceable-selector
              selectedServiceable=fi.value
              onSelect=(action 'setServiceable' fi.update)
              onClear=(action 'clearServiceable')
              handleScan=(action 'handleScan' fi.update)
       }}
      </div>
    {{else}}
  {{/unless}}
{{/f.input}}

{{#f.input name='name' as |fi|}}
  <div class="required field">
    <label>Reason</label>
    <input type="text" value={{fi.value}} onchange={{action fi.update value="target.value"}} autocomplete="off">
  </div>
{{/f.input}}
```

```hbs
{{f.submit label=(t (if serviceOrder.isNew 'saveAndStartWork' 'update')) disabled=f.loading}}
```

**Comment by czosel** [comment]
Hi @cmackenz, that sounds like a tough one! Do you think you could create a reproduction in a twiddle? That would make analyzing this much easier. You could start from [this twiddle](https://ember-twiddle.com/3691a8576c35ff149bfc26a564ec5437).

**Comment by utilityboy** [comment]
@cmackenz, I believe the issue is that the custom control has an action named `update`.  Changing the action name in the twiddle to something else seems to work around the problem.

**Comment by czosel** [comment]
@utilityboy Hmm, I didn't manage to fix it by renaming the action - can you share the modified twiddle?

@cmackenz Thanks for the twiddle, that helps a lot :smiley:

**Comment by utilityboy** [comment]
@czosel, I was just coming back to mention that didn't work for us as well.  Still an issue.

**Comment by anehx** [comment]
Hi @cmackenz 

I took a look at it and found the problem: your button in the custom component does not explicitly set a `type` property. By spec, the default `type` on a `<button>` is `submit`. This means if you press enter in a `<form>` it will search for the nearest button with type `submit` and "press" it to submit the form.

**TL;DR**
To fix your problem, set `type="button"` on your button in the custom component.

**Comment by cmackenz** [comment]
wow, good catch! Thank you
